x-pack/plugins/cases/public/components/create/form.tsx

```diff
@@ -76,130 +45,70 @@ export interface CreateCaseFormProps extends Pick<Partial<CreateCaseFormFieldsPr
-            </div>
-            <div css={containerCss(euiTheme)} />
-          </>
-        ),
-      }),
-      [
-        isSubmitting,
-        euiTheme,
-        caseAssignmentAuthorized,
-        canShowCaseSolutionSelection,
-        availableOwners,
-        draftStorageKey,
-      ]
-    );
-
-    const secondStep = useMemo(
-      () => ({
-        title: i18n.STEP_TWO_TITLE,
-        children: (
-          <div>
-            <SyncAlertsToggle isLoading={isSubmitting} />
-          </div>
-        ),
-      }),
-      [isSubmitting]
-    );
-
-    const thirdStep = useMemo(
-      () => ({
-        title: i18n.STEP_THREE_TITLE,
-        children: (
-          <div>
-            <Connector
-              connectors={connectors}
-              isLoadingConnectors={isLoadingConnectors}
-              isLoading={isSubmitting}
-            />
-          </div>
-        ),
-      }),
-      [connectors, isLoadingConnectors, isSubmitting]
-    );
-
-    const allSteps = useMemo(
-      () => [firstStep, ...(isSyncAlertsEnabled ? [secondStep] : []), thirdStep],
-      [isSyncAlertsEnabled, firstStep, secondStep, thirdStep]
+    const shouldShowOwnerSelector = Boolean(!owner.length && availableOwners.length > 1);
+    const { reset } = useFormContext();
+
+    const { data: connectors = [], isLoading: isLoadingConnectors } =
+      useGetSupportedActionConnectors();
+
+    const onOwnerChange = useCallback(
+      (newOwner: string) => {
+        onSelectedOwner(newOwner);
+        reset({
+          resetValues: true,
+          defaultValue: getInitialCaseValue({
+            owner: newOwner,
+            connector: currentConfiguration.connector,
+          }),
+        });
+      },
+      [currentConfiguration.connector, onSelectedOwner, reset]
     );
 
     return (
       <>
-        {isSubmitting && (
-          <EuiLoadingSpinner
-            css={css`
-              position: absolute;
-              top: 50%;
-              left: 50%;
-              z-index: 99;
-            `}
-            data-test-subj="create-case-loading-spinner"
-            size="xl"
-          />
-        )}
-        {withSteps ? (
-          <EuiSteps
-            headingElement="h2"
-            steps={allSteps}
-            data-test-subj={'case-creation-form-steps'}
+        {shouldShowOwnerSelector && (
+          <CreateCaseOwnerSelector
+            selectedOwner={selectedOwner}
+            availableOwners={availableOwners}
+            isLoading={isLoadingCaseConfiguration}
+            onOwnerChange={onOwnerChange}
           />
-        ) : (
-          <>
-            {firstStep.children}
-            {isSyncAlertsEnabled && secondStep.children}
-            {thirdStep.children}
-          </>
         )}
+        <CreateCaseFormFields
+          connectors={connectors}
+          isLoading={isLoadingConnectors || isLoadingCaseConfiguration}
+          withSteps={withSteps}
+          draftStorageKey={draftStorageKey}
```

Comment:
I think this line is the source of [this bug](https://github.com/elastic/kibana/issues/187236#issuecomment-2199831049).

We could make the `draftStorageKey` optional and not include it in the creation form. What do you guys think?

[The draft was originally implemented for the comments in the edit case page.](https://github.com/elastic/kibana/pull/146327)

Reply:
yes it's fine to not have description storage while creating a case. It is only useful for edit description scenario.

Reply:
What is the current behavior on main? What if you spent a lot of time writing the description on the case and you made a mistake (refresh the browser, etc)?

Reply:
Yes, let's do this! It can be done on another PR.